pkg/monitor/lease.go

```diff
@@ -86,16 +122,204 @@ func leaseVIP(cfgPath, masterDevice, name string) error {
 		return err
 	}
 
-	cmd := exec.Command("dhclient", "-d", "--no-pid", "-sf", "/bin/true",
-		"-lf", getLeaseFile(cfgPath, name), "-v", iface.Name, "-H", name)
+	leaseFile := GetLeaseFile(cfgPath, name)
+	if f, err := os.Create(leaseFile); err != nil {
+		log.WithFields(logrus.Fields{
+			"name": leaseFile,
+		}).WithError(err).Error("Failed to create lease file")
+		return err
+	} else {
+		f.Close()
+	}
+
+	watcher, err := createLeaseFileWatcher(log, leaseFile)
+
+	if err != nil {
+		log.WithFields(logrus.Fields{
+			"filename": leaseFile,
+		}).WithError(err).Error("Failed to create a watcher for lease file")
+		return err
+	}
+
+	// -sf avoiding dhclient from setting the received IP to the interface
+	// --no-pid in order to allow running multiple dhclients
+	commandArgs := []string{"-v", iface.Name, "-H", name,
+		"-sf", "/bin/true", "-lf", leaseFile, "-d",
+		"--no-pid", "-pf", fmt.Sprintf("/var/run/dhclient.%s.pid", iface.Name)}
+
+	cmd := exec.Command("dhclient", commandArgs...)
 	cmd.Stderr = os.Stderr
 
-	return cmd.Start()
+	if !runInfinite {
```

Comment:
Why is this configurable? AFAICT we only ever call this with true. It seems like a whole bunch of dead code.